### DISCUSSION TITLE
Is the code of fetching VS Code extension data outdated?

### DISCUSSION THREAD

**Discussion by leodevbro** [badges/shields#8759] [General] [answered]
Here:

https://github.com/badges/shields/blob/f67251a9cae8ec1f84bbe6e9fe42bc6e0c7f0a90/services/visual-studio-marketplace/visual-studio-marketplace-base.js#L83

I see the source link for the number 402 (0x192) is this:

https://github.com/microsoft/vscode/blob/main/src/vs/platform/extensionManagement/common/extensionGalleryService.ts

But I cannot find `0x192` in that `extensionGalleryService.ts` file.

Is the file `visual-studio-marketplace-base.js` a little outdated?

In `extensionGalleryService.ts` I see:
```typescript
enum Flags {
  None = 0x0,
  IncludeVersions = 0x1,
  IncludeFiles = 0x2,
  IncludeCategoryAndTags = 0x4,
  IncludeSharedAccounts = 0x8,
  IncludeVersionProperties = 0x10,
  ExcludeNonValidated = 0x20,
  IncludeInstallationTargets = 0x40,
  IncludeAssetUri = 0x80,
  IncludeStatistics = 0x100,
  IncludeLatestVersionOnly = 0x200,
  Unpublished = 0x1000,
  IncludeNameConflictInfo = 0x8000,
}

```

And I cannot find `0x192`.

And, the strange thing is that 402 (0x192) still works fine, maybe because the VS Code Marketplace API supports backward compatibility?

**Comment by chris48s** [comment, marked as answer]
As noted in the comment, 0x192 does not correspond to a single value - it is a combined hex value representing multiple flags.
